cron_server_maintenance/models/maintenance.py

```diff
@@ -0,0 +1,42 @@
+from odoo import models, fields, api, _
+import requests
+import json
+from datetime import datetime
+
+
+class Maintenance(models.Model):
+    _inherit = 'maintenance.equipment'
+
+    is_monitored = fields.Boolean(string="Is Monitored")
+    monitor_url = fields.Char(string="Monitor URL")
+    monitor_is_running = fields.Selection([('offline', 'Offline'), ('done', 'Running')], string='Monitor',
+                                          default='offline', tracking=True)
+    monitor_log_ids = fields.One2many('maintenance.equipment.server.log', 'maintenance_id', string="Logs",
+                                      readonly=True)
+
+    def server_auto_sync(self):
+        servers = self
+        if not servers:
```

Comment:
How does this work?
On the row above servers = self
If self is None or False then how can we do self.env?

Reply:
okay, so I did that first, because of the cron job and the smart button on the form view. Cron job is in a loops through all records, but the servers = self will make the smart button also work.

self will return a record when a button is clicked from the form view, but will return None on cron